src/testing/basic_test.rs

```diff
@@ -1,390 +1,230 @@
-                    )
-                    .await
-                }
-                _ => {
-                    let previous_justify_qc =
-                        sqc_msgs[(i - 1) as usize].proposal.data.justify_qc.clone();
-                    // metadata
-                    let _metadata = <TestBlockHeader as BlockHeader<TestTypes>>::metadata(
-                        &sqc_msgs[(i - 1) as usize].proposal.data.block_header,
-                    );
-                    let leaf =
-                        Leaf::from_quorum_proposal(&sqc_msgs[(i - 1) as usize].proposal.data);
-
-                    let q_data = QuorumData::<TestTypes> {
-                        leaf_commit: leaf.commit(),
-                    };
+        let da_proposal = Arc::new(DaProposalMessage {
+            view_number: ViewNumber::new(round as u64),
+            txn_commitments,
+            num_nodes: NUM_STORAGE_NODES,
+            sender: pub_key,
+            builder_commitment: block_builder_commitment.clone(),
+        });
 
-                    let previous_qc_view_number =
-                        sqc_msgs[(i - 1) as usize].proposal.data.view_number.u64();
-                    let view_number = if previous_qc_view_number == 0
-                        && previous_justify_qc.view_number.u64() == 0
-                    {
-                        ViewNumber::new(0)
-                    } else {
-                        ViewNumber::new(1 + previous_justify_qc.view_number.u64())
-                    };
-                    // form a justify qc
-                    SimpleCertificate::<TestTypes, QuorumData<TestTypes>, SuccessThreshold> {
-                        data: q_data.clone(),
-                        vote_commitment: q_data.commit(),
-                        view_number,
-                        signatures: previous_justify_qc.signatures.clone(),
-                        _pd: PhantomData,
-                    }
+        let block_header = TestBlockHeader {
+            block_number: round as u64,
+            payload_commitment: block_vid_commitment,
+            builder_commitment: block_builder_commitment,
+            timestamp: round as u64,
+        };
+
+        let justify_qc = match prev_quorum_proposal.take() {
+            None => {
+                QuorumCertificate::<TestTypes>::genesis(
+                    &TestValidatedState::default(),
+                    &TestInstanceState::default(),
+                )
+                .await
+            }
+            Some(prev_proposal) => {
+                let prev_justify_qc = &prev_proposal.justify_qc;
+                let prev_view_number = prev_proposal.view_number.u64();
+                let view_number = if prev_view_number == 0 && prev_justify_qc.view_number.u64() == 0
+                {
+                    ViewNumber::new(0)
+                } else {
+                    ViewNumber::new(1 + prev_justify_qc.view_number.u64())
+                };
```

Comment:
Looks like `view_number` will never be 1. Since we've handled the genesis case in the `None` clause above, I'm wondering whether we should remove the `ViewNumber::new(0)` case here.

Reply:
Yeah, this can just be `ViewNumber::new(round as u64)`